kfs-purap/src/main/java/edu/arizona/kfs/module/purap/service/impl/PurapAccountingServiceImpl.java

```
@@ -0,0 +1,147 @@
+    
+    @Override
+    public List<PurApAccountingLine> generateAccountDistributionForProration(List<SourceAccountingLine> accounts, KualiDecimal totalAmount, Integer percentScale, Class clazz, List<PurApItem> purApItems) {
+        String methodName = "generateAccountDistributionForProration()";
+        if ( LOG.isDebugEnabled() ) {
+            LOG.debug(methodName + " started");
+        }
+                
+        if(totalAmount.isZero() || isAllSummaryAccoountsAmountZero(accounts)) {
+            return getFirstItemAccountsForDistribution(purApItems, clazz);
+        } else {
+            return super.generateAccountDistributionForProration(accounts, totalAmount, percentScale, clazz);
+        }
+    }
+
+    /*
+     * Check if all the summary accounts amount are zero. If all are zero, it will cause exception when creating
+     * distribution account.
+     */
+    private boolean isAllSummaryAccoountsAmountZero(List<SourceAccountingLine> summaryAccounts) {
+    	boolean isZero = true;
+    	for (SourceAccountingLine account : summaryAccounts) {
+    		isZero = account.getAmount().isZero();
+    		if (!isZero) {
+    			break;
+    		}
+    	}
+    	return isZero;
+    	
+    }
+
+    /*
+     * get first line item's accounts as distribution accounts
+     */
+    private List<PurApAccountingLine> getFirstItemAccountsForDistribution(List<PurApItem> purapItems, Class clazz) {
+        List<PurApAccountingLine> newAccounts = new ArrayList<PurApAccountingLine>();
+    	
+        for (PurApItem item : purapItems) {
+            if (item.getItemType().isLineItemIndicator()) {
+            	for (PurApAccountingLine account : item.getSourceAccountingLines()) {
+                    PurApAccountingLine newAccountingLine = null;
+                    try {
+                        newAccountingLine = (PurApAccountingLine) clazz.newInstance();
+                    }
+                    catch (InstantiationException e) {
+                        e.printStackTrace();
```

I'm not a fan of printing STEs directly to STDOUT -- I'd suggest either log out at an appropriate level, or swallow the exception entirely with a comment explaining why that's not a bad thing (e.g.: // ObjectPopulationUtils can handle a this nulled out reference... I have no idea if that's the case).